src/Microsoft.ML.Data/Transforms/KeyToVectorTransform.cs

```diff
@@ -206,393 +187,605 @@ public override void Save(ModelSaveContext ctx)
+            }
+
+            private JToken SaveAsPfaCore(BoundPfaContext ctx, int iinfo, ColInfo info, JToken srcToken)
+            {
+                Contracts.AssertValue(ctx);
+                Contracts.Assert(0 <= iinfo && iinfo < _infos.Length);
+                Contracts.Assert(_infos[iinfo] == info);
+                Contracts.AssertValue(srcToken);
+                Contracts.Assert(CanSavePfa);
+
+                int keyCount = info.TypeSrc.ItemType.KeyCount;
+                Host.Assert(keyCount > 0);
+                // If the input type is scalar, we can just use the fanout function.
+                if (!info.TypeSrc.IsVector)
+                    return PfaUtils.Call("cast.fanoutDouble", srcToken, 0, keyCount, false);
+
+                JToken arrType = PfaUtils.Type.Array(PfaUtils.Type.Double);
+                if (_parent._bags[iinfo] || info.TypeSrc.ValueCount == 1)
+                {
+                    // The concatenation case. We can still use fanout, but we just append them all together.
+                    return PfaUtils.Call("a.flatMap", srcToken,
+                        PfaContext.CreateFuncBlock(new JArray() { PfaUtils.Param("k", PfaUtils.Type.Int) },
+                        arrType, PfaUtils.Call("cast.fanoutDouble", "k", 0, keyCount, false)));
+                }
+
+                // The bag case, while the most useful, is the most elaborate and difficult: we create
+                // an all-zero array and then add items to it.
+                const string funcName = "keyToVecUpdate";
+                if (!ctx.Pfa.ContainsFunc(funcName))
+                {
+                    var toFunc = PfaContext.CreateFuncBlock(
+                        new JArray() { PfaUtils.Param("v", PfaUtils.Type.Double) }, PfaUtils.Type.Double,
+                        PfaUtils.Call("+", "v", 1));
+
+                    ctx.Pfa.AddFunc(funcName,
+                        new JArray(PfaUtils.Param("a", arrType), PfaUtils.Param("i", PfaUtils.Type.Int)),
+                        arrType, PfaUtils.If(PfaUtils.Call(">=", "i", 0),
+                        PfaUtils.Index("a", "i").AddReturn("to", toFunc), "a"));
+                }
+
+                return PfaUtils.Call("a.fold", srcToken,
+                    PfaUtils.Call("cast.fanoutDouble", -1, 0, keyCount, false), PfaUtils.FuncRef("u." + funcName));
+            }
+
+            private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, ColInfo info, string srcVariableName, string dstVariableName)
+            {
+                string opType = "OneHotEncoder";
+                var node = ctx.CreateNode(opType, srcVariableName, dstVariableName, ctx.GetNodeName(opType));
+                node.AddAttribute("cats_int64s", Enumerable.Range(1, info.TypeSrc.ItemType.KeyCount).Select(x => (long)x));
+                node.AddAttribute("zeros", true);
+                return true;
+            }
+        }
+    }
+
+    public sealed class KeyToVectorEstimator : IEstimator<KeyToVectorTransform>
```

Comment:
>IEstimator [](start = 47, length = 10)

we agreed to make this estimator a trivial estimator, and not enforce the match between train-time schema and transform-time schema